### ISSUE TITLE
Using os.path.join would make this a bit more friendly to Windows

### ISSUE THREAD

**Issue by scottbilas** [srstevenson/xdg#7]
Not all of the XDG vars make sense on Windows, but the ones that default to starting at $HOME could benefit from not hard-coding the slash type.

A small suggestion:

```
diff --git 1/xdg.py "2/C:\\Users\\scott\\dotfiles\\updot\\.venv\\Lib\\site-packages\\xdg.py"
index 478fcba..95900cf 100644
--- 1/xdg.py
+++ "2/C:\\Users\\scott\\dotfiles\\updot\\.venv\\Lib\\site-packages\\xdg.py"
@@ -57,12 +57,12 @@ def _getenv(variable: str, default: str) -> str:
     return os.environ.get(variable) or default


-XDG_CACHE_HOME = _getenv('XDG_CACHE_HOME', os.path.expandvars('$HOME/.cache'))
+XDG_CACHE_HOME = _getenv('XDG_CACHE_HOME', os.path.expandvars(os.path.join('$HOME', '.cache')))
 XDG_CONFIG_DIRS = _getenv('XDG_CONFIG_DIRS', '/etc/xdg').split(':')
 XDG_CONFIG_HOME = _getenv('XDG_CONFIG_HOME',
-                          os.path.expandvars('$HOME/.config'))
+                          os.path.expandvars(os.path.join('$HOME', '.config')))
 XDG_DATA_DIRS = _getenv('XDG_DATA_DIRS',
                         '/usr/local/share/:/usr/share/').split(':')
 XDG_DATA_HOME = _getenv('XDG_DATA_HOME',
-                        os.path.expandvars('$HOME/.local/share'))
+                        os.path.expandvars(os.path.join('$HOME', '.local', 'share')))
 XDG_RUNTIME_DIR = os.getenv('XDG_RUNTIME_DIR')
```

**Comment by srstevenson** [comment]
`xdg` follows the specification strictly, which specifies the directory separator as `/` and doesn't describe different behaviour on Windows. That said, your suggestion is pragmatic and doesn't change the behaviour on Unix. Are there many Windows applications that follow the specification?

**Comment by scottbilas** [comment]
I don't know about "many" Windows applications, but some native ports such as ssh and vim put at least their config files as dot folders under $HOME (unsure whether they pay attention to XDG_* env vars as well).

**Comment by srstevenson** [comment]
Thanks @scottbilas, I've released [3.0.0](https://github.com/srstevenson/xdg/releases/tag/3.0.0) to PyPI with your change.
